fusillade/utils/authorize.py

```diff
@@ -23,13 +25,25 @@ def get_policy_statement(evaluation_results: List[Dict[str, Any]], policies: Lis
     """
     for evaluation_result in evaluation_results:
         for ms in evaluation_result['MatchedStatements']:
-            policy_index = int(ms['SourcePolicyId'].split('.')[-1]) - 1
             begin = ms['StartPosition']['Column']
             end = ms['EndPosition']['Column']
-            policy = policies[policy_index]
-            ms['statement'] = policy['policy_document'][begin:end]
-            ms['SourcePolicyId'] = policy['name']
-            ms['SourcePolicyType'] = policy['type']
+            try:
+                policy_index = int(ms['SourcePolicyId'].split('.')[-1]) - 1
+            except ValueError:
+                if resource_policy:
+                    ms['statement'] = resource_policy[begin:end]
+                    ms['SourcePolicyId'] = 'ResourcePolicy'
+                    ms['SourcePolicyType'] = 'ResourcePolicy'
+                else:
+                    logging.warning({"msg": "Failed to parse evaluation response.",
+                                     "matched_statement": ms,
```

Comment:
Should this key be "MatchedStatements " like the evalutation results? it would be nice to keep them all the same :)